src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -1020,6 +1020,14 @@ public override void VisitMetadataDeclarationSyntax(MetadataDeclarationSyntax sy
         public override void VisitVariableDeclarationSyntax(VariableDeclarationSyntax syntax)
             => AssignTypeWithDiagnostics(syntax, diagnostics =>
             {
+                if (syntax.Type is {})
```

Comment:
[very nit picky]: Using `is {}` as a null check is kinda hard to read, esp. for newcomers to C# (`is not null` feels much more approachable).